qiskit/quantum_info/synthesis/two_qubit_decompose.py

```diff
@@ -547,6 +547,208 @@ def specialize(self):
         self.K2r = np.asarray(RYGate(k2rtheta)) @ np.asarray(RXGate(k2rlambda))
 
 
+class TwoQubitControlledUDecomposer:
+    """Decompose two-qubit unitary in terms of a desired U ~ Ud(α, 0, 0) ~ Ctrl-U gate
+    that is locally equivalent to an RXXGate."""
+
+    _default_1q_basis = "ZYZ"
+
+    def __init__(self, unitary, rxx_equivalent_gate: Type[Gate]):
+        """Initialize the KAK decomposition.
+
+        Args:
+            unitary: The unitary to be decomposed.
+            rxx_equivalent_gate: Gate that is locally equivalent to an RXXGate:
+            U ~ Ud(α, 0, 0) ~ Ctrl-U gate.
+        """
+        self.rxx_equivalent_gate = rxx_equivalent_gate
```

Comment:
```suggestion
        # Do a few spot checks that the KAK decomposition gives (x, 0, 0)
        for test_angle in [0.2, 0.3, np.pi / 2]:
            decomp = TwoQubitWeylDecomposition(Operator(rxx_equivalent_gate(test_angle)).data)
            if not np.allclose([decomp.b, decomp.c], [0, 0]):
                raise QiskitError(f"{rxx_equivalent_gate.__name__} is not equivalent to an RXXGate.")

        self.rxx_equivalent_gate = rxx_equivalent_gate
```
Since `Operator(gate(Parameter("angle")).data)` does not work testing a few angles is the next best thing I could come up with. Open to better suggestions but I think this should catch most cases where the gate is not equivalent to a `RXXGate`.

Reply:
Yes. If profiling shows that this test is a performance bottleneck, then we could make a static property on the common gates to mark them as rxx-equivalent and only fallback to this numerical check for unusual gates who don't have that property.